src/library_manager/lib_manager.c

```diff
@@ -572,13 +661,38 @@ int lib_manager_register_module(const uint32_t component_id)
 		goto cleanup;
 	}
 
-	/* Fill the new_drv_info structure with already known parameters */
-	/* Check already registered components */
 	mod = (struct sof_man_module *)((const uint8_t *)desc + SOF_MAN_MODULE_OFFSET(entry_index));
 	const struct sof_uuid *uid = (struct sof_uuid *)&mod->uuid[0];
 
 	lib_manager_prepare_module_adapter(drv, uid);
 
+	build_info = (const struct sof_module_api_build_info *)((const char *)desc -
+		     SOF_MAN_ELF_TEXT_OFFSET + mod->segment[SOF_MAN_SEGMENT_TEXT].file_offset);
+
+	tr_info(&lib_manager_tr,
+		"lib_manager_register_module(): Module API version: %u.%u.%u, format: 0x%x",
+		build_info->api_version_number.fields.major,
+		build_info->api_version_number.fields.middle,
+		build_info->api_version_number.fields.minor,
+		build_info->format);
+
+	/* Check if module is native */
+	if (build_info->format == SOF_MODULE_API_BUILD_INFO_FORMAT &&
+	    build_info->api_version_number.full == SOF_MODULE_API_CURRENT_VERSION) {
+		/* Use lib_manager shim functions */
+		drv->ops.create = &lib_manager_module_create;
+		drv->ops.free = &lib_manager_module_free;
```

Comment:
do I understand it correctly, that after this change modules.c will only be used for IADK? That would actually be good indeed.

Reply:
By design IADK modules are inheriting interface from native loadable modules.

Reply:
`modules.c` (part of the Processing Module Adapter) was supposed to be responsible only for calling C++ functions related to IADK modules. After this change it is no longer used by other module types.